app/docker/views/images/import/importImageController.js

```diff
@@ -1,28 +1,63 @@
+import { PorImageRegistryModel } from 'Docker/models/porImageRegistry';
+
 angular.module('portainer.docker').controller('ImportImageController', [
   '$scope',
   '$state',
   'ImageService',
   'Notifications',
   'HttpRequestHelper',
-  function ($scope, $state, ImageService, Notifications, HttpRequestHelper) {
+  'Authentication',
+  'ImageHelper',
+  'endpoint',
+  function ($scope, $state, ImageService, Notifications, HttpRequestHelper, Authentication, ImageHelper, endpoint) {
     $scope.state = {
       actionInProgress: false,
     };
 
+    $scope.endpoint = endpoint;
+
+    $scope.isAdmin = Authentication.isAdmin();
+
     $scope.formValues = {
       UploadFile: null,
       NodeName: null,
+      RegistryModel: new PorImageRegistryModel(),
     };
 
+    $scope.setPullImageValidity = setPullImageValidity;
+    function setPullImageValidity(validity) {
+      $scope.state.pullImageValidity = validity;
+    }
+
+    function tagImage(id) {
+      const registryModel = $scope.formValues.RegistryModel;
+      if (registryModel.Image) {
+        const image = ImageHelper.createImageConfigForContainer(registryModel);
+        ImageService.tagImage(id, image.fromImage).catch(function error(err) {
```

Comment:
use async/await with try catch.

Reply:
I returned the promise as we are awaiting for a response in that.

Reply:
done